incontext.py

```diff
@@ -150,13 +209,14 @@ def add_command(self, name, function, help=""):
         @param help: The help string that describes the command to be printed when the user passes the '--help' flag.
         @type help: str
         """
+        self._plugins.add_plugin(PLUGIN_TYPE_COMMAND, name, wrap_add_command(name, function, help))
+        
+    def _add_command(self, name, function, help):
```

Comment:
It should be possible to move this wrapped functionality into a method on a new `CommandPlugin` object to further remove the indirection.